File: Python/Problems/problem_2.py
#!/usr/bin/env python3


<<<<<<< HEAD:Problems/Problem_2/Python/problem_2.py
=======
def solution():
    UPPER = 4000000
    fib_sum = sum_even_fibonacci(UPPER)
    return f'The sum of even Fibonacci numbers below {UPPER} is {fib_sum}'


>>>>>>> refactorProblemRunners:Python/Problems/problem_2.py
def sum_even_fibonacci(upperbound):
    result = 2

    previous = 1
    current = 2

    while previous + current < upperbound + 1:
        current, previous = current + previous, current

        if not current % 2:
            result += current

    return result

<<<<<<< HEAD:Problems/Problem_2/Python/problem_2.py

if __name__ == '__main__':
    UPPERBOUND = 4000000
    fib_sum = sum_even_fibonacci(4000000)
    print(f'Sum of even Fibonacci numbers below {UPPERBOUND} is {fib_sum}')

=======
>>>>>>> refactorProblemRunners:Python/Problems/problem_2.py

File: Rust/src/problems/problem_2.rs
pub fn solution() -> String {
    let upper = 4000000;
    let sum = sum_even_fibonacci(upper);
    format!("The sum of even Fibonacci numbers below {} is {}", upper, sum)
}


fn sum_even_fibonacci(upper: usize) -> usize {
    let mut sum = 2;

    let mut previous = 1;
    let mut current = 2;

    while previous + current < upper + 1 {
        current += previous;
        previous = current - previous;

        if current % 2 == 0 {
            sum += current;
        }
    }

    sum
}


#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_sum_even_fibonacci() {
        assert_eq!(sum_even_fibonacci(10), 10);
    }
}
